Add birthday-tracker plugin

Raboro opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/Raboro/Obsidian-Birthday-Tracker-Plugin

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["description": "This plugin allows you to keep track of all birthdays of your family and friends.",](https://github.com/Raboro/Obsidian-Birthday-Tracker-Plugin/blob/528b70d17f1eb36b34a2f5b83c59edec08095bb5/manifest.json#L6)
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.
Avoid including sentences like "This is a plugin that does xy" in your description.

[id: 'birthday-tracker-track-birthdays',](https://github.com/Raboro/Obsidian-Birthday-Tracker-Plugin/blob/528b70d17f1eb36b34a2f5b83c59edec08095bb5/src/main.ts#L20)
Obsidian will automatically prefix the command id with the plugin id,
you don't need to include it yourself.

[this.app.workspace.detachLeavesOfType(BIRTHDAY_TRACKER_VIEW_TYPE);](https://github.com/Raboro/Obsidian-Birthday-Tracker-Plugin/blob/528b70d17f1eb36b34a2f5b83c59edec08095bb5/src/main.ts#L42)
Instead of detaching I would recommend to check first If a view already exist and reveal that instead, this is related to: https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60> 

[for (const file of this.app.vault.getFiles()) {](https://github.com/Raboro/Obsidian-Birthday-Tracker-Plugin/blob/528b70d17f1eb36b34a2f5b83c59edec08095bb5/src/main.ts#L49-L54)
Make the file that contains birthday data configurable.
The way it's implemented currently is very bad for performance in bigger vaults.

Raboro commented:
Thank you very much for the quick feedback. I fixed the description as suggested in the guidelines, removed the plugin ID from the command and checked if the view already exists instead of separating it, already fixed.

For my poor performance of the loop iterating over all files, I would add a setting to add the path of the file with the birthday data and use ``app.vault.getAbstractFileByPath()`` instead as suggested in the guidelines.

And for the birthday file configuration. I have plans to make multiple data collections possible. First as now, second by a table. Maybe metadata and from multiple files later.

Does it fit like this?
